library/src/lapack/roclapack_potrf.hpp

```diff
@@ -136,443 +136,195 @@ void rocsolver_potrf_rightlooking_getMemorySize(const rocblas_int n,
                         shiftA + idx2D(j + jb, j + jb, lda), lda, strideA, batch_count);
-
-                    if(istat != rocblas_status_success)
-                    {
-                        return (istat);
-                    }
                 }
+                j += nb;
             }
-            j += nb;
         }
-    }
-    else
-    {
-        // Compute the Cholesky factorization A = L*L'.
-        while(j < n - POTRF_POTF2_SWITCHSIZE(T))
+        else
         {
-            // Factor diagonal and subdiagonal blocks
-            jb = std::min(n - j, nb); // number of columns in the block
-            ROCSOLVER_LAUNCH_KERNEL(reset_info, gridReset, threads, 0, stream, iinfo, batch_count, 0);
-            rocsolver_potf2_template<T>(handle, uplo, jb, A, shiftA + idx2D(j, j, lda), lda,
-                                        strideA, iinfo, batch_count, scalars, (T*)work1, pivots);
-
-            // test for non-positive-definiteness.
-            ROCSOLVER_LAUNCH_KERNEL(chk_positive<U>, gridReset, threads, 0, stream, iinfo, info,
-                                    j + row_offset, batch_count);
-
-            if(j + jb < n)
+            // Compute the Cholesky factorization A = L*L'.
+            while(j < n - POTRF_POTF2_SWITCHSIZE(T))
             {
+                // Factor diagonal and subdiagonal blocks
+                jb = std::min(n - j, nb); // number of columns in the block
+                ROCSOLVER_LAUNCH_KERNEL(reset_info, gridReset, threads, 0, stream, iinfo,
+                                        batch_count, 0);
+                rocsolver_potf2_template<T>(handle, uplo, jb, A, shiftA + idx2D(j, j, lda), lda,
+                                            strideA, iinfo, batch_count, scalars, (T*)work1, pivots);
+
+                // test for non-positive-definiteness.
+                ROCSOLVER_LAUNCH_KERNEL(chk_positive<U>, gridReset, threads, 0, stream, iinfo, info,
+                                        j, batch_count);
+
+                if(j + jb < n)
                 {
-                    auto const istat = rocsolver_trsm_lower<BATCHED, STRIDED, T>(
+                    // update trailing submatrix
+                    rocsolver_trsm_lower<BATCHED, STRIDED, T>(
                         handle, rocblas_side_right, rocblas_operation_conjugate_transpose,
                         rocblas_diagonal_non_unit, (n - j - jb), jb, A, shiftA + idx2D(j, j, lda),
                         lda, strideA, A, shiftA + idx2D(j + jb, j, lda), lda, strideA, batch_count,
                         optim_mem, work1, work2, work3, work4);
 
-                    if(istat != rocblas_status_success)
-                    {
-                        return (istat);
-                    }
-                }
-
-                {
-                    // update trailing submatrix
-                    auto const istat = rocblasCall_syrk_herk<BATCHED, T>(
+                    rocblasCall_syrk_herk<BATCHED, T>(
```

Comment:
should there be something like ROCBLAS_CHECK() to check the return status for all rocblas calls?

Reply:
Sure, I'll add that in